config/tests/samples/create/samples.go

```diff
@@ -90,17 +90,44 @@ func getNamespaces(samples []Sample) []string {
 	return results
 }
 
-func RunCreateDeleteTest(t *Harness, unstructs []*unstructured.Unstructured, cleanupResources bool) {
+type CreateDeleteTestOptions struct {
+	// Create is the set of objects to create
+	Create []*unstructured.Unstructured
+
+	// Updates is the set of objects to update (after all objects have been created)
+	Updates []*unstructured.Unstructured
+
+	// CleanupResources is true if we should delete resources when we are done
+	CleanupResources bool
+}
+
+func RunCreateDeleteTest(t *Harness, opt CreateDeleteTestOptions) {
+	ctx := t.Ctx
+
 	// Create and reconcile all resources & dependencies
-	for _, u := range unstructs {
-		if err := t.GetClient().Create(context.TODO(), u); err != nil {
+	for _, u := range opt.Create {
+		if err := t.GetClient().Create(ctx, u); err != nil {
 			t.Fatalf("error creating resource: %v", err)
 		}
 	}
-	waitForReady(t, unstructs)
-	// Clean up resources on success or if cleanupResources flag is true
-	if cleanupResources {
-		cleanup(t, unstructs)
+
+	waitForReady(t, opt.Create)
+
+	if len(opt.Updates) != 0 {
+		// treat as a patch
+		for _, updateUnstruct := range opt.Updates {
+			if err := t.GetClient().Patch(ctx, updateUnstruct, client.Apply, client.FieldOwner("kcc-tests"), client.ForceOwnership); err != nil {
+				t.Fatalf("error updating resource: %v", err)
+			}
+		}
+		time.Sleep(5 * time.Second)
```

Comment:
:( This seems like a flake waiting to happen. Can we switch to something which stops waiting early when success is reached but has a longer timeout for failure? I know K8s has utilities like that. Not saying you have to use those utilities but ....

Reply:
Fair - it's not ideal.  We need status.observedGeneration to do this "right" (that's why we're waiting here).  I'll check out if we have that field today....

Reply:
Good news, we do have observedGeneration and it looks easy to improve our wait-for-ready logic to incorporate it: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/871

Reply:
I've removed this sleep.  It's not 100% safe until we get #871 in, but it should "fail open" I believe